examples/dac.rs

```diff
@@ -0,0 +1,73 @@
+#![no_main]
+#![no_std]
+
+use panic_semihosting as _;
+
+use cortex_m::asm;
+
+use cortex_m_rt::entry;
+use stm32f3xx_hal::{
+    dac::{Dac, DacBits, Trigger},
+    pac,
+    prelude::*,
+};
+
+#[entry]
+fn main() -> ! {
+    // Set up microcontroller peripherals
+    let dp = pac::Peripherals::take().unwrap();
+
+    // Set up clocks
+    let mut flash = dp.FLASH.constrain();
+    let mut rcc = dp.RCC.constrain();
+    let _clocks = rcc.cfgr.freeze(&mut flash.acr);
+
+    // Set up the DAC output pin.
+    // The output pin you choose selects the respective DAC channel:
+    //  - PA4: channel 1
+    //  - PA5: channel 2
+    let mut gpioa = dp.GPIOA.split(&mut rcc.ahb);
+    let dac_pin = gpioa.pa4.into_analog(&mut gpioa.moder, &mut gpioa.pupdr);
+
+    // Set up the DAC
+    let reference_voltage = 2.915;
```

Comment:
@strom-und-spiele Where is that voltage value coming from?

Reply:
Ah, I see you got that from experimentation. Is this a value we can expect to be the same on each user of an F3DISCOVERY? If not, I think we should document how a user can find there own correct value. There must be some way to measure/infer the VREF for a given board, no?

Reply:
I wasn't so sure either and checked the schematics of the stm32f3discovery again.

VREF is a stabilized version VDD.
VDD is directly connected via Jumper 3 to 3V
3V is powered by the 5V supply and adjusted by a [BAT60JFILM](https://www.mouser.de/datasheet/2/389/bat60-1848979.pdf) which has a voltage drop that depends on the power being used, so it's roughly 3V. Less if your chip is running something more demanding.

I suggest
* replacing the 2.9… by a solid 3
* add to the docs the info that the ref value should be known by the schematics and can be roughly determined by setting `set_value` to 255 and measuring the output
* add a hint to the docs in the example